tedana/workflows/tedana.py

```diff
@@ -677,13 +706,17 @@ def tedana_workflow(
             )
             ica_selector = selection.automatic_selection(comptable, n_echos, n_vols, tree=tree)
             n_likely_bold_comps = ica_selector.n_likely_bold_comps
-            if (n_restarts < maxrestart) and (n_likely_bold_comps == 0):
-                LGR.warning("No BOLD components found. Re-attempting ICA.")
-            elif n_likely_bold_comps == 0:
-                LGR.warning("No BOLD components found, but maximum number of restarts reached.")
-                keep_restarting = False
-            else:
+
+            if ica_method=='robustica':  #########BTBTBT
                 keep_restarting = False
+            else:               
+                if (n_restarts < maxrestart) and (n_likely_bold_comps == 0):
+                    LGR.warning("No BOLD components found. Re-attempting ICA.")
+                elif n_likely_bold_comps == 0:
+                    LGR.warning("No BOLD components found, but maximum number of restarts reached.")
```

Comment:
This might require greater user feedback; they may not understand the consequence of this sequence for them.

Reply:
It is written by the tedana team, so I will it to them.